src/coprocessor/codec/mysql/time.rs

```diff
@@ -387,6 +576,163 @@ impl Time {
+                }
+            }
+            'm' => Ok(format!("{:02}", self.time.month())),
+            'c' => Ok(format!("{}", self.time.month())),
+            'D' => Ok(format!(
+                "{}{}",
+                self.time.day(),
+                abbr_day_of_month(self.time.day() as i32)
+            )),
+            'd' => Ok(format!("{:02}", self.time.day())),
+            'e' => Ok(format!("{}", self.time.day())),
+            'j' => Ok(format!("{:03}", self.time.days())),
+            'H' => Ok(format!("{:02}", self.time.hour())),
+            'k' => Ok(format!("{}", self.time.hour())),
+            'h' | 'I' => {
+                let t = self.time.hour();
+                if t == 0 || t == 12 {
+                    Ok("12".to_string())
+                } else {
+                    Ok(format!("{:02}", t % 12))
+                }
+            }
+            'l' => {
+                let t = self.time.hour();
+                if t == 0 || t == 12 {
+                    Ok("12".to_string())
+                } else {
+                    Ok(format!("{}", t % 12))
+                }
+            }
+            'i' => Ok(format!("{:02}", self.time.minute())),
+            'p' => {
+                let hour = self.time.hour();
+                if (hour / 12) % 2 == 0 {
+                    Ok("AM".to_string())
+                } else {
+                    Ok("PM".to_string())
+                }
+            }
+            'r' => {
+                let h = self.time.hour();
+                if h == 0 {
+                    Ok(format!(
+                        "{:02}:{:02}:{:02} AM",
+                        12,
+                        self.time.minute(),
+                        self.time.second()
+                    ))
+                } else if h == 12 {
+                    Ok(format!(
+                        "{:02}:{:02}:{:02} PM",
+                        12,
+                        self.time.minute(),
+                        self.time.second()
+                    ))
+                } else if h < 12 {
+                    Ok(format!(
+                        "{:02}:{:02}:{:02} AM",
+                        h,
+                        self.time.minute(),
+                        self.time.second()
+                    ))
+                } else {
+                    Ok(format!(
+                        "{:02}:{:02}:{:02} PM",
+                        h - 12,
+                        self.time.minute(),
+                        self.time.second()
+                    ))
+                }
+            }
+            'T' => Ok(format!(
+                "{:02}:{:02}:{:02}",
+                self.time.hour(),
+                self.time.minute(),
+                self.time.second()
+            )),
+            'S' | 's' => Ok(format!("{:02}", self.time.second())),
+            'f' => Ok(format!("{:06}", self.time.nanosecond() / 1000)),
+            'U' => {
+                let w = self.time.week(false, true, true);
```

Comment:
In Go `U` use week mode `0`, so the argument should be `false, true, false`?

Reply:
TiDB uses `Week()`:

```go
func (t mysqlTime) Week(mode int) int {
	if t.month == 0 || t.day == 0 {
		return 0
	}
	_, week := calcWeek(&t, weekMode(mode))
	return week
}
```

where weekMode:

```go
func weekMode(mode int) weekBehaviour {
	var weekFormat weekBehaviour
	weekFormat = weekBehaviour(mode & 7)
	if (weekFormat & weekBehaviourMondayFirst) == 0 {
		weekFormat ^= weekBehaviourFirstWeekday
	}
	return weekFormat
}
```

From the code we can see that if `mondayFirst` is not set (1st arg == false), `firstWeekday` is always set (3rd arg <- true).